src/main/resources/lib/guillotine/util/factory.js

```diff
@@ -76,5 +76,273 @@ function processBooleanFilter(inputFilter) {
+        }
+        if (expression.language) {
+            dslExpression.language = expression.language;
+        }
+        if (expression.boost) {
+            dslExpression.boost = expression.boost;
+        }
+        holder.stemmed = dslExpression;
+    }
+}
+
+function createDslBooleanExpression(holder, expression) {
+    if (expression) {
+        let dslExpression = {};
+        if (expression.should) {
+            dslExpression.should = [];
+            (expression.should || []).forEach(boolExpression => dslExpression.should.push(createDslQuery(boolExpression)))
+        } else if (expression.must) {
+            dslExpression.must = [];
+            (expression.must || []).forEach(boolExpression => dslExpression.must.push(createDslQuery(boolExpression)))
+        } else if (expression.mustNot) {
+            dslExpression.mustNot = [];
+            (expression.mustNot || []).forEach(boolExpression => dslExpression.mustNot.push(createDslQuery(boolExpression)))
+        } else if (expression.filter) {
+            dslExpression.filter = [];
+            (expression.filter || []).forEach(boolExpression => dslExpression.filter.push(createDslQuery(boolExpression)))
+        } else {
+            throw 'Must be set property for boolean expression';
+        }
+        holder.boolean = dslExpression;
+    }
+}
+
+function extractPropertyValues(graphQLValue) {
+    if (graphQLValue.stringValues) {
+        return graphQLValue.stringValues;
+    }
+    if (graphQLValue.doubleValues) {
+        return graphQLValue.doubleValues;
+    }
+    if (graphQLValue.longValues) {
+        return graphQLValue.longValues;
+    }
+    if (graphQLValue.booleanValues) {
+        return graphQLValue.booleanValues;
+    }
+    if (graphQLValue.localDateValues) {
+        return graphQLValue.localDateValues;
+    }
+    if (graphQLValue.localDateTimeValues) {
+        return graphQLValue.localDateTimeValues;
+    }
+    if (graphQLValue.localTimeValues) {
+        return graphQLValue.localTimeValues;
+    }
+    if (graphQLValue.instantValues) {
+        return graphQLValue.instantValues;
+    }
+    throw 'Values must be not null';
+}
+
+function extractPropertyValue(graphQLValue) {
+    if (graphQLValue.stringValue) {
+        return graphQLValue.stringValue;
+    }
+    if (graphQLValue.doubleValue != null) {
+        return graphQLValue.doubleValue;
+    }
+    if (graphQLValue.longValue != null) {
+        return graphQLValue.longValue;
+    }
+    if (graphQLValue.booleanValue != null) {
+        return graphQLValue.booleanValue;
+    }
+    if (graphQLValue.localDateValue) {
+        return graphQLValue.localDateValue;
+    }
+    if (graphQLValue.localDateTimeValue) {
+        return graphQLValue.localDateTimeValue;
+    }
+    if (graphQLValue.localTimeValue) {
+        return graphQLValue.localTimeValue;
+    }
+    if (graphQLValue.instantValue) {
+        return graphQLValue.instantValue;
+    }
+    throw 'Value must be not null';
+}
```

Comment:
So we're happy with this? Lets say someone creates a range with lessthan string, greaterthan double? I'm assuming the validator will make a nice error message?

Reply:
I added validation for that